src/CreeDictionary/CreeDictionary/paradigm/manager.py

```diff
@@ -30,21 +32,50 @@ def static_paradigm_for(self, name: str) -> Optional[Paradigm]:
         """
         return self._name_to_layout.get(name)
 
+    def dynamic_paradigm_for(
+        self, *, lemma: str, word_class: str
+    ) -> Optional[Paradigm]:
+        """
+        Returns a dynamic paradigm for the given lemma and word class.
+        Returns None if no such paradigm can be generated.
+        """
+        if layout := self._wc_to_layout.get(word_class):
+            return self._inflect(layout, lemma)
+
+        # No matching word class means no paradigm:
+        return None
+
     def _load_static_from(self, path: Path):
         """
-        Loads all .tsv files in the path as "static" paradigms.
+        Loads all .tsv files in the path as static paradigms.
+        """
+        for filename, layout in self._load_all_layouts_in_directory(path):
+            self._name_to_layout[filename.stem] = layout.as_static_paradigm()
+
+    def _load_dynamic_from(self, path: Path):
+        """
+        Loads all .tsv files as dynamic layouts.
+        """
+        for filename, layout in self._load_all_layouts_in_directory(path):
+            self._wc_to_layout[filename.stem] = layout
+
+    def _inflect(self, layout: ParadigmLayout, lemma: str) -> Paradigm:
+        """
+        Given a layout and a lemma, produce a paradigm with forms generated by the FST.
         """
+        template2analysis = layout.generate_fst_analyses(lemma=lemma)
+        analysis2forms = self._generator.bulk_lookup(list(template2analysis.values()))
+        template2forms = {
+            template: analysis2forms[analysis]
+            for template, analysis in template2analysis.items()
+        }
+        return layout.fill(template2forms)
 
+    @staticmethod
+    def _load_all_layouts_in_directory(path: Path):
         for layout_file in path.glob("*.tsv"):
             layout = ParadigmLayout.loads(layout_file.read_text(encoding="UTF-8"))
```

Comment:
Do you know any reason to explicitly specify the encoding these days?

Reply:
Genuinely curious, I never specify this and wonder if I should be.

Reply:
It's a habit I use because Python uses the system encoding as the default (yuck!). Although this generally isn't a problem on most modern systems, this can cause problems when the system's locales are b0rked (it happens), current locale is something you don't expect, or when you're running on _legacy systems_ (see: Windows before Windows 10—I was writing code on Windows 7 for my current job, and this line is _necessary_ for that use case 🙃 ).